plugin/validator_test.go

```diff
@@ -177,6 +177,47 @@ func TestValidator_LoadSchema(t *testing.T) {
 		assert.Empty(t, pluginName)
 		assert.NotNil(t, err)
 	})
+	t.Run("overrides ngx.re.match and ngx.re.find", func(t *testing.T) {
+		schema, err := os.ReadFile("testdata/override-ngx-re-test.lua")
+		assert.Nil(t, err)
+		pluginName, err := v.LoadSchema(string(schema))
+		assert.EqualValues(t, "override-ngx-re-test", pluginName)
+		assert.Nil(t, err)
+
+		pluginFind := `{
+			"name": "override-ngx-re-test",
+			"config": {
+				"findstr": "foo, 1234"
+			},
+			"enabled": true,
+			"protocols": ["http"]
+		}`
+
+		pluginMatch := `{
+			"name": "override-ngx-re-test",
+			"config": {
+				"matchstr": "bar"
+			},
+			"enabled": true,
+			"protocols": ["http"]
+		}`
+
+		pluginBoth := `{
+			"name": "override-ngx-re-test",
+			"config": {
+				"findstr": "foo, 1234",
+				"matchstr": "bar"
+			},
+			"enabled": true,
+			"protocols": ["http"]
+		}`
+
+		testCases := []string{pluginFind, pluginMatch, pluginBoth}
+		for _, testCase := range testCases {
+			err = v.Validate(testCase)
+			assert.Nil(t, err)
+		}
+	})
```

Comment:
- Let's isolate this into it's own test; doesn't need to be with the `LoadSchema` tests.
- Add more robust tests; see schema update suggestions

Something like this (test is not complete not tested)
```
func TestValidator_NginxRegex(t *testing.T) {
	schema, err := os.ReadFile("testdata/override-ngx-re-test.lua")
	require.NoError(t, err)
	require.NotNil(t, schema)

	v, err := NewValidator(ValidatorOpts{})
	require.NoError(t, err)
	require.NotNil(t, v)
	pluginName, err := v.LoadSchema(string(schema))
	require.NoError(t, err)
	require.Equal(t, "override-ngx-re-test", pluginName)

	tests := []struct {
		name        string
		config      string
		expectedErr string
	}{
		{
			name: "ensure ngx.re.find is loaded and contains subject",
			config: `"find_regex": "([0-9])",
				"subject": "A123"`,
		},
		{
			name: "ensure ngx.re.find is loaded and fails to contain subject",
			config: `"find_regex": "([a-z][0-9]+)",
				"subject": "A123"`,
			expectedErr: "failed to find subject A123 using regex ([a-z][0-9]+)",
		},
		{
			name: "ensure ngx.re.match is loaded and matches subject",
			config: `"match_regex": "([A-Z][0-9]+)",
				"subject": "A123"`,
		},
		{
			name: "ensure ngx.re.match is loaded and fails to match subject",
			config: `"match_regex": "([a-z][0-9]+)",
				"subject": "A123"`,
			expectedErr: "failed to match subject A123 using regex ([a-z][0-9]+)",
		},
	}

	for _, test := range tests {
		plugin := fmt.Sprintf(`{
				"name": "ngx-re-test",
				"config": {
					%s
				},
				"enabled": true,
				"protocols": ["http"]
			}`, test.config)
		err = v.Validate(plugin)
		if len(test.expectedErr) > 0 {
			require.ErrorContains(t, err, test.expectedErr)
		} else {
			require.NoError(t, err)
		}
	}
}
```